src/accountsdb/index.zig

```diff
@@ -1058,7 +1060,81 @@ pub const DiskMemoryAllocator = struct {
     }
 };
 
-test "tests disk allocator on hashmaps" {
+test "swissmap resize" {
+    var map = SwissMap(Pubkey, AccountRef, pubkey_hash, pubkey_eql).init(std.testing.allocator);
+    defer map.deinit();
+
+    try map.ensureTotalCapacity(100);
+
+    const ref = AccountRef.default();
+    map.putAssumeCapacity(Pubkey.default(), ref);
+
+    // this will resize the map with the key still in there
+    try map.ensureTotalCapacity(200);
+    const get_ref = map.get(Pubkey.default()) orelse return error.MissingAccount;
+    try std.testing.expect(std.meta.eql(get_ref, ref));
+}
+
+test "account index update/remove reference" {
+    const allocator = std.testing.allocator;
+
+    var index = try AccountIndex.init(allocator, allocator, 8);
+    defer index.deinit(true);
+    try index.ensureTotalCapacity(100);
+
+    // pubkey -> a
+    var ref_a = AccountRef.default();
+    index.indexRef(&ref_a);
+
+    var ref_b = AccountRef.default();
+    ref_b.slot = 1;
+    index.indexRef(&ref_b);
+
+    // make sure indexRef works
+    {
+        var ref_head_rw = index.getReference(&ref_a.pubkey).?;
+        const ref_head, var ref_head_lg = ref_head_rw.writeWithLock();
+        ref_head_lg.unlock();
+        _, const ref_max = ref_head.highestRootedSlot(10);
+        try std.testing.expect(ref_max == 1);
```

Comment:
We'll get nicer & more informative error messages on test failures if we use `std.testing.expectEqual`:
```suggestion
        try std.testing.expectEqual(1, ref_max);
```
Note: if this habit developed due to historical misbehavior of `expectEqual` with literals as the first argument, you'll be glad to know that the odd behavior has largely been resolved - the only unhappy case is when testing the tag of a tagged union, which has to be written as:
```zig
try std.testing.expectEqual(.Foo, std.meta.activeTag(U{ .Foo = value }));
```

Reply:
eb28c8e

yeah you guessed it - bad habit from the prev `expectEqual` - thanks for catching